pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -305,23 +305,36 @@ func maybeRetryDownload(ctx *downloaderContext,
 			status.Key())
 		return
 	}
-	status.LastErr = ""
-	status.LastErrTime = time.Time{}
-	status.RetryCount += 1
+	status.ClearErrorInfo()
 	// XXX do we need to adjust reservedspace??
 
 	dst, errStr := lookupDatastoreConfig(ctx, config.DatastoreID, config.Name)
 	if dst == nil {
-		status.LastErr = errStr
-		status.LastErrTime = time.Now()
+		status.SetErrorInfo(errStr)
 		publishDownloaderStatus(ctx, status)
 		return
 	}
+
+	// try to reserve storage
+	// must be released on error
+	kb := types.RoundupToKB(config.Size)
+	if !tryReserveSpace(ctx, status, kb) {
+		errStr := fmt.Sprintf("Would exceed remaining space. "+
```

Comment:
Why are you not printing the ErrorStr here?

Reply:
Addressed, extracted the common functionality to HandleDownlodFail()